pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/LoaderUtils.java

```diff
@@ -90,4 +97,28 @@ public static void writeIndexToV3Format(SegmentDirectory.Writer segmentWriter, S
     }
     return stringList;
   }
+
+  /**
+   * Try to recover a segment from reload failure, no effect for normal segments.
+   * <p>Should be called before trying to load the segment.
```

Comment:
should be called before trying to load the segment OR trying to load the metadata right?